### DISCUSSION TITLE
Tap method and overlaid elements on Android.

### DISCUSSION THREAD

**Discussion by iMaupin** [wix/Detox#3356] [Q&A]
Is there a way to have detox only tap an element if it is not overlayed? This does not tend to be an issue on iOS because overlays affect an element's visibility. This, however, has not proven to be the case when I run my test suite on android.

The current behavior in these cases is that detox taps the element on top of the desired element, consequently, breaking tests that should pass.

Does anybody have any suggestions for addressing this behavior?

**Comment by d4vidi** [comment]
@iMaupin the [`toBeVisible()`](https://wix.github.io/Detox/docs/api/expect/#tobevisible) assertion may come in handy with this, as it accepts a minimal visible-surface threshold.

**Reply by iMaupin** [reply]
@d4vidi Thanks for the quick reply. Generally, when I experience this problem such an assertion has already been made. The issue is that only on my android testing devices does a visibility check set to 100 pass when the element is overlaid.

Usually, the overlaying element is a navigation bottom bar or header, and when I scroll to match the 100 percent visible the needed element remains below the header or footer, yet still passes the assertion.
